activation/wire/wire_v2.go

```diff
@@ -196,8 +196,7 @@ func (mc *MarriageCertificate) Root() []byte {
 // MerkleProofV2 proves membership of multiple challenges in a PoET membership merkle tree.
 type MerkleProofV2 struct {
 	// Nodes on path from leaf to root (not including leaf)
-	Nodes       []types.Hash32 `scale:"max=32"`
-	LeafIndices []uint64       `scale:"max=256"` // support merging up to 256 IDs
+	Nodes []types.Hash32 `scale:"max=32"`
```

Comment:
At that point we should probably calculate the merkle root of the `Nodes` in `MerkleProofV2` and add it to the ATX tree instead of the scale encoded proof. Wdyt?

Encoding speed wise it should make little difference, but we can then refer to it if we need it at some point.

Reply:
I doubt we will need to refer to a specific node in this merkle proof ever. This proof is either valid or not.

> Encoding speed wise it should make little difference

There should be a difference because creating a merkle proof of it requires up to 31 hashes while encoding - 1 hash.

Reply:
🤷 we could benchmark it, I just have the feeling at some point in the future we might want to change it and then we have to parse blobs again 😅

Reply:
How shall we proceed with this?

Reply:
I benchmarked the two approaches with the following benchmark code:
```go
func Benchmark_MembershipRoot(b *testing.B) {
	membership := MerkleProofV2{
		Nodes: make([]types.Hash32, 32),
	}
	for i := range membership.Nodes {
		membership.Nodes[i] = types.RandomHash()
	}
	b.ResetTimer()

	b.Run("encoding", func(b *testing.B) {
		for range b.N {
			r := codec.MustEncode(&membership)
			require.NotEmpty(b, r)
		}
	})
	b.Run("merkle proof", func(b *testing.B) {
		for range b.N {
			tree, err := merkle.NewTreeBuilder().WithHashFunc(atxTreeHash).Build()
			require.NoError(b, err)

			for _, n := range membership.Nodes {
				tree.AddLeaf(n.Bytes())
			}

			r := tree.Root()
			require.NotEmpty(b, r)
		}
	})
}
```

Results:
```
Benchmark_MembershipRoot/encoding-20         	 2088924	       543.7 ns/op	    1225 B/op	       3 allocs/op
Benchmark_MembershipRoot/merkle_proof-20     	  185016	      6633 ns/op	    1816 B/op	      49 allocs/op
```

Reply:
It is slower but only by ~ 6 µs. Doing this for every ATX in an epoch (~ 5 Mio at the moment) adds 30 seconds of processing time per epoch. I think that's negligible, wdyt?

Reply:
It's also allocating 16x more, stressing the GC more.

I can't think of a possible use case for the merkle tree here. Perhaps we should ask research about it?

Anyway, I don't think this PR must be blocked by this. We can still change how ATXv2 ID is calculated.